### ISSUE TITLE
PHP test setup is a poor developer experience

### ISSUE THREAD

**Issue by aaemnnosttv** [google/site-kit-wp#1445]
Currently our PHP tests and supporting scripts yield a bit of a poor experience for new developers.

There are multiple scripts in the repo for installing the tests and there isn't a clear difference as to which script to use.

* [`install-wp-tests.sh`](https://github.com/google/site-kit-wp/blob/46290f84ed6897f18c094e57949fcddeac7e7587/tests/bin/install-wp-tests.sh)  
Looks familiar and is based on the [template provided by wp-cli](https://github.com/wp-cli/scaffold-command/blob/master/templates/install-wp-tests.sh) by the same name. However, the script is modified and slightly out of date.  
Most importantly, it does not install WordPress core itself. It also does not include the option to skip database creation. It also has different default value for `$WP_TESTS_DIR`. This is the script we use in Travis, where we clone the `wordpress-develop` repo beforehand, although I'm not sure why this is really necessary.
* [`install.sh`](https://github.com/google/site-kit-wp/blob/46290f84ed6897f18c094e57949fcddeac7e7587/tests/bin/install.sh)  
This appears to be the same as the [template provided by wp-cli](https://github.com/wp-cli/scaffold-command/blob/master/templates/install-wp-tests.sh) although slightly out of date.
* [`run-wp-unit-tests.sh`](https://github.com/google/site-kit-wp/blob/46290f84ed6897f18c094e57949fcddeac7e7587/tests/bin/run-wp-unit-tests.sh)  
This seems fairly straightforward by the name, but it actually runs `install.sh` internally and has hardcoded values for all arguments passed to that script except the `$WP_VERSION`

#### A few things should happen for sure
* Consolidate `install-wp-tests.sh` and `install.sh` into a single script  
_Delete `install.sh` and synchronize `install-wp-tests.sh` with the latest version of the script from the wp-cli repo_  
    - This eliminates the confusion by using the same script by the same name that many developers are already familiar with
* Delete `run-wp-unit-tests.sh` as it's too unflexible and isn't necessary  
_Running tests should be as simple as running `composer test`_

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* There should only be a single script for installing tests:  
`tests/bin/install-wp-tests.sh`
* `install-wp-tests.sh` should be an exact copy of the [template provided by wp-cli](https://github.com/wp-cli/scaffold-command/blob/master/templates/install-wp-tests.sh) by the same name
* `tests/bin/install.sh` should be removed
* `tests/bin/run-wp-unit-tests.sh` should be removed
* `.travis.yml` should be updated to no longer clone the wordpress-develop repo  
    - This will no longer be necessary using the standard `install-wp-tests.sh` script  
    https://github.com/google/site-kit-wp/blob/46290f84ed6897f18c094e57949fcddeac7e7587/.travis.yml#L61-L65
    - Definitions of `WP_DEVELOP_DIR` and `WP_TESTS_DIR` should also be unnecessary

## Implementation Brief

* Merge #1446 

## QA Brief

* Ensure old files are removed
* Ensure local set up with `tests/bin/run-wp-unit-tests.sh` works as expected
* All tests should continue to pass

## Changelog entry

* N/A

**Comment by felixarntz** [comment]
IB ✅

**Comment by aaemnnosttv** [comment]
There's one tiny change needed here since this was merged after #1883 as the `allowed_failures` definition for the WP nightly job no longer matches. I'll create a quick follow-up PR here to fix.

**Comment by eugene-manuilov** [comment]
QA :heavy_check_mark: Old scripts are deleted, new script works, tests work too.
